app/controllers/admin/petition_details_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Admin::PetitionDetailsController < Admin::AdminController
+  respond_to :html
+  before_action :fetch_petition
+
+  def show
+  end
+
+  def update
+    if @petition.update_attributes(petition_params)
+      flash[:notice] = 'Petition has been successfully updated'
+      redirect_to edit_admin_petition_path(@petition)
+    else
+      render :show
+    end
+  end
+
+  private
+
+  def fetch_petition
+    @petition = Petition.find(params[:petition_id])
+    raise ActiveRecord::RecordNotFound unless @petition.in_todo_list?
```

Comment:
Minor thing, but if we do:

```ruby
def fetch_petition
  @petition = Petition.todo_list.find(params[:petition_id])
end
```

We don't need the guard clause at the end as AR will raise the exception for us by going through the scope.